test/api/mocha/utils/testUtils.js

```diff
@@ -30,20 +34,96 @@ const executeRequest = async (url, method, token, body = null) => {
+  // Determine the actual file path based on the createNewMetricsFiles flag
+  let actualOutputFile = outputMetricsResponsesFile
+  if (createNewMetricsFiles) {
+    // Extract directory and filename
+    const lastSlashIndex = outputMetricsResponsesFile.lastIndexOf('/')
+    const lastBackslashIndex = outputMetricsResponsesFile.lastIndexOf('\\')
+    const separatorIndex = Math.max(lastSlashIndex, lastBackslashIndex)
+    
+    if (separatorIndex >= 0) {
+      const dir = outputMetricsResponsesFile.substring(0, separatorIndex + 1)
+      const file = outputMetricsResponsesFile.substring(separatorIndex + 1)
+      actualOutputFile = `${dir}new-${file}`
+    } else {
+      // No directory in the path, just prepend 'new-'
+      actualOutputFile = `new-${outputMetricsResponsesFile}`
+    }
+  }
+ 
+  // Read existing file to preserve all data
+  let metricsData = {}
+  try {
+    const fileContent = readFileSync(actualOutputFile, 'utf8')
+    metricsData = JSON.parse(fileContent)
+  } catch (err) {
+    console.log(`Error creating new metrics file or parsing existing file: ${err.message}`)
+    // Continue with empty object if file doesn't exist or parsing fails
+  }
+  
+  // Update metrics data with new test case data
   if (!metricsData[testCaseName]) {
     metricsData[testCaseName] = {}
   }
   metricsData[testCaseName][username] = responseData
-  writeFileSync(metricsFilePath, JSON.stringify(metricsData, null, 2), 'utf8')
+  
+  // Write back to file as JSON
+  writeFileSync(actualOutputFile, JSON.stringify(metricsData, null, 2), 'utf8')
 }
 
+/**
+ * Conditionally outputs metrics data based on the STIGMAN_SAVE_METRICS_DATA environment variable
+ * Works for both regular metrics and meta metrics
+ * @param {string} testCaseName - The test case name
+ * @param {string} username - The username
+ * @param {Object} responseData - The response data to save
+ * @param {string} outputJsonFile - Path to the output file
+ */
+const conditionalMetricsOutput = (testCaseName, username, responseData, outputJsonFile) => {
+  // Only record metrics if the environment variable is set
+  if (saveMetricsData) {
+    outputMetricsToJSON(testCaseName, username, responseData, outputJsonFile)
+  }
+}
+
+
+
 const getUUIDSubString = (length = 20) => {
   return uuidv4().substring(0, length)
 }
 
+/**
+ * Export application data as JSONL
+ * @param {object} options - Options for exporting
+ * @param {string} options.format - Format of the exported data (default: 'jsonl')
+ * @returns {Promise<string>} - The exported data
+ */
+const exportAppData = async (options = {}) => {
+  const format = options.format || 'jsonl'
+  
+  const response = await fetch(`${baseUrl}/op/appdata?elevate=true&format=${format}`, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${adminToken}`
+    }
+  })
+  
+  if (!response.ok) {
+    throw new Error(`Failed to export data: ${response.status} ${response.statusText}`)
+  }
+  
+  return response.text()
+}
+
```

Comment:
This exported function is called only by an unused function (`loadAndExportAllAppData`) and should be removed.